assets/js/sections/contesting.js

```diff
@@ -208,20 +214,35 @@ $('#start_date').change(function () {
 
 // On Key up check and suggest callsigns
 $("#callsign").keyup(function () {
-	var call = $(this).val();
+	var call = $(this).val().toUpperCase();
 	if (call.length >= 3) {
 
-		$.ajax({
-			url: 'lookup/scp',
-			method: 'POST',
-			data: {
-				callsign: $(this).val().toUpperCase()
-			},
-			success: function (result) {
-				$('.callsign-suggestions').text(result);
-				highlight(call.toUpperCase());
-			}
-		});
+        if (scp_data.status < 0 || scp_data.request != call.substr(0, scp_data.request.length)) {
```

Comment:
A few things here, you should use the `!==` comparison so it doesnt try to do any type interpretation.

see here: https://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons

---

The logic check on `scp_data.status < 0` will only ever be true on page load?

---

> scp_data.request != call.substr(0, scp_data.request.length)

I think this can be cleaned up with `startsWith`

```
if (scp_data.request.startsWith(call)) {
  // Your code here
}
```

https://www.w3schools.com/jsref/jsref_startswith.asp

Reply:
:) Strangely enough, I used `startsWith` further down in the code. Thanks (or sorry for me being... well, distracted by other languages ;) ).

-1: Yes.